taskiq/cli/worker/run.py

```diff
@@ -153,10 +153,12 @@ def run_worker(args: WorkerArgs) -> None:  # noqa: WPS213
 
     :raises ValueError: if reload flag is used, but dependencies are not installed.
     """
-    logging.basicConfig(
-        level=logging.getLevelName(args.log_level),
-        format="[%(asctime)s][%(name)s][%(levelname)-7s][%(processName)s] %(message)s",
-    )
+    if args.configure_logging:
+        logging.basicConfig(
```

Comment:
I think this detaches the logs from the root logger but then I can't control what level `taskiq` should emit, right?
I could achieve that by doing something like...
```python
logging.getLogger("taskiq").setLevel(<level>)
```
...in my application.

But this is super counter-intuitive, specially because we'd still have `--log-level` as an argument.

From my perspective, we should still be able to control the log level from the cli and `taskiq` should simply detaches its loggers from root.

Reply:
I implemented this [here](https://github.com/taskiq-python/taskiq/compare/0f1fda2d87d5580450708312bf65534f60db8c32..ca5d3fe0711ed3033760ed62c1b7361eb6c09a2b). 

Still, I don't get the idea of detaching logger from others. Imagine you have loguru configured. In can intercept log calls, but still you would have to manually add taskiq in list of intercepted loggers, because it's not propogated to the root logger. I'm not quite sure if it's okay.

Is it a problem for your case, if we do not detach `taskiq` named logger from the root one?

Reply:
For my application is not a concern because I decided to set `propagate=False` in my handlers.

I've never used `loguru` so for me it's hard to say. But from your description I think it makes more sense to explicitly add an "interception" than explicitly disabling propagation from my logger on my application.

`Taskiq` is just an implementation detail (even being the most important dependency - don't get me wrong :)) for my application so I think it makes more sense that my application should "say" if `taskiq` logs should appear than having `taskiq` impacting my logs. I have the same train of thought for a new dependency: I wouldn't like to have to remember that I have to disable its loggers but rather I'd like to be able to easily enable if I want it.

On top of that, by configuring the root logger you also "force" that loggers from other dependencies to also show up, because not all of them explicitly set `propagate` to `False` - meaning that the stream is prone to "pollution" since loggers from all other packages will start to be propagated to the `root` logger. This (potentially, I'm not entirely sure) also means that I lose granularity over the levels of those other loggers: I can't easily set the level of `taskiq` logs to `INFO` and `ddtrace` to `ERROR`, for example, because by setting `taskiq` logs to `INFO` it actually means `root` logs to `INFO` and because `ddtrace_logger.propagate = True` I think I'll also receive logs when `ddtrace_logger.level = INFO`. 

And, to be honest, I've never seen a dependency configure the root logger (let's take a look at uvicorn [here](https://github.com/encode/uvicorn/blob/master/uvicorn/config.py#L74C1-L74C15) and [here](https://github.com/encode/uvicorn/blob/master/uvicorn/config.py#L379), for instance).

Does all of this make sense?

Reply:
Yes it makes sense. But the main idea is that taskiq has a solid and ready-to-use configuration that would work for any project and you can change every aspect of it with few configuration parameters. It only configures logger for its own processes, it doesn't affect client-size, because it run this code only on worker processes. It configures root logger, because in many cases people don't configure logging. This is an out of the box simple configuration that you can use easily. If you want to configure logging by yourself, you can turn off automatic configuration and do whatever you want. 

If it doesn't make sense to you, please make a request with your solution that works. We would continue discussion there. Thank you for finding time for a discussion.

Reply:
Yes, being able to completely turn on and off logging configuration also makes sense. 

I would only disagree with "it doesn't affect client-size": although I acknowledge that this is true for most applications, it's not the case for many others - i.e. those larger systems whose services are decoupled and communication are done trough messaging. For those systems, the client side is the worker processes as the application entrypoint is that task that will process the message published in the related broker.

Reply:
Okay. But even in this case when worker processes are clients. How would it affect them if there would be no logging configuration? If you have an Idea of how you'd like this feature to be implemented, please provide me with simple example. Because I like the Idea and don't see edge cases in current implementation. If you do, please tell me what you see and maybe how to fix it.

Reply:
Oh it was just a comment. I think this solution and your train of thought is great.

Reply:
Thank you very much for the discussion. It was really enjoyable. I will merge this pull request. But still, you can raise issues any time.